src/scenic/core/vectors.py

```diff
@@ -4,427 +4,659 @@
-	@decorator.decorator
-	def wrapper(wrapped, instance, *args, **kwargs):
-		def helper(*args, **kwargs):
-			if any(needsSampling(arg) for arg in itertools.chain(args, kwargs.values())):
-				return VectorMethodDistribution(method, instance, args, kwargs)
-			elif any(needsLazyEvaluation(arg)
-			         for arg in itertools.chain(args, kwargs.values())):
-				# see analogous comment in distributionFunction
-				return makeDelayedFunctionCall(helper, args, kwargs)
-			else:
-				return wrapped(instance, *args, **kwargs)
-		return helper(*args, **kwargs)
-	return wrapper(method)
+    """Decorator for methods that produce vectors. See distributionMethod."""
+    @functools.wraps(method)
+    def helper(self, *args, **kwargs):
+        if any(needsSampling(arg) for arg in itertools.chain(args, kwargs.values())):
+            return VectorMethodDistribution(method, self, args, kwargs)
+        elif any(needsLazyEvaluation(arg)
+                 for arg in itertools.chain(args, kwargs.values())):
+            # see analogous comment in distributionFunction
+            return makeDelayedFunctionCall(helper, (self,) + args, kwargs)
+        else:
+            return method(self, *args, **kwargs)
+    return helper
+
+class Orientation:
+    """An orientation in 3D space."""
+    def __init__(self, rotation):
+        if not isinstance(rotation, Rotation):
+            raise ValueError("Orientation's 'rotation' parameter must be a SciPy rotation."\
+                " Perhaps you want to use a constructor?")
+        self.r = rotation
+        self.q = rotation.as_quat()
+
+    @classmethod
+    def fromQuaternion(cls, quaternion) -> Orientation:
+        r = Rotation.from_quat(quaternion)
+        return cls(r)
+
+    @classmethod
+    @distributionFunction
+    def fromEuler(cls, yaw, pitch, roll) -> Orientation:
+        r = Rotation.from_euler('ZXY', [yaw, pitch, roll], degrees=False)
+        return cls(r)
+
+    @property
+    def w(self) -> float:
+        return self.q[3]
+
+    @property
+    def x(self) -> float:
+        return self.q[0]
+
+    @property
+    def y(self) -> float:  
+        return self.q[1]
+
+    @property
+    def z(self) -> float: 
+        return self.q[2]
+
+    @property
+    def yaw(self) -> float:
+        """Yaw in the global coordinate system."""
+        return self.eulerAngles[0]
+
+    @property
+    def pitch(self) -> float:
+        """Pitch in the global coordinate system."""
+        return self.eulerAngles[1]
+
+    @property
+    def roll(self) -> float:
+        """Roll in the global coordinate system."""
+        return self.eulerAngles[2]
+
+    @staticmethod
+    def _coerce(thing) -> Orientation:
+        if isinstance(thing, Orientation):
+            return thing
+        elif hasattr(thing, 'toOrientation'):
+            return thing.toOrientation()
+        elif isinstance(thing, Vector):
+            return Orientation.fromEuler(*thing)
```

Comment:
What's the motivation for this case? Seems like it just lets type errors go by silently.